### ISSUE TITLE
Access violation on SSL_shutdown 

### ISSUE THREAD

**Issue by illera88** [nmap/nmap#1365]
Hi,

I statically compiled `ncat` with OPENSSL and it's failing when doing the following:

I open a cmd and listen with:
`ncat.exe -lvp 4444 --ssl`

In the debugger I start ncat with:
`ncat.exe 127.0.0.1 4444 -e cmd.exe --ssl -v`.

The connection gets established but when I type some command to get:

```
Unhandled exception at 0x00530056 in ncat.exe: 0xC0000005: Access violation executing location 0x00530056.
```
at `SSL_shutdown(info->fdn.ssl);` at line 366 in `ncat_exec_win.c`.

**Comment by illera88** [comment]
I've been debugging it and the problem is in the infinite loop at `subprocess_thread_func`. 

For some weird reason (if I debug with step over in VS it works fine) `WaitForMultipleObjects` returns 0 and when calling to `GetLastError` the error gotten is 996:
```
ERROR_IO_INCOMPLETE

996 (0x3E4)

Overlapped I/O event is not in a signaled state.
```

**Comment by dmiller-nmap** [comment]
Thanks for the bug report. I think what happened was the fix for #1267 involved adding a call to `SSL_shutdown` and `SSL_free` just after the loop, so when this one is encountered, it's a double-free. I changed it to also set the `fdn.ssl` pointer to `NULL` so that this call is skipped, which ought to solve your problem. I don't think we can take the earlier calls out entirely, because they are there to avoid SSL errors during more of the connection teardown process.

**Comment by dmiller-nmap** [comment]
Sorry, I forgot to push the change up. Watch for it within the next hour, then retest. Thanks!
